File: project/graphdata/model/antifraud/af_addrloan.py
# -*- coding: utf-8 -*-
from lib.application import db
from .base import Base

class AfAddrloan(db.Model, Base):
    __bind_key__ = 'xhh_antifraud'
    __tablename__ = 'af_addr_loan'

    id = db.Column(db.BigInteger, primary_key=True)
    request_id = db.Column(db.BigInteger, nullable=False,index=True)
    aid = db.Column(db.Integer)
    user_id = db.Column(db.BigInteger, nullable=False,index=True)
    user_total = db.Column(db.Integer, nullable=False)
    loan_all = db.Column(db.Integer, nullable=False)
    overdue_norepay = db.Column(db.Integer, nullable=False)
    overdue_repay = db.Column(db.Integer, nullable=False)
    overdue7_norepay = db.Column(db.Integer, nullable=False)
    overdue7_repay = db.Column(db.Integer, nullable=False)
    loan_total = db.Column(db.Integer, nullable=False)
    last_loan_day = db.Column(db.Integer, nullable=False)
    normal_repay = db.Column(db.Integer, nullable=False)
    realadl_tot_reject_num = db.Column(db.Integer, nullable=True)
    realadl_tot_freject_num = db.Column(db.Integer, nullable=True)
    realadl_tot_sreject_num = db.Column(db.Integer, nullable=True)
    realadl_tot_dlq14_num = db.Column(db.Integer, nullable=True)
    realadl_dlq14_ratio = db.Column(db.Numeric(12, 2),nullable=True)
    history_bad_status = db.Column(db.Integer, nullable=True)
    create_time = db.Column(db.DateTime, nullable=False)

File: project/strategy/models/antifraud/AfAddrLoan.php
<?php

namespace app\models\antifraud;

use Yii;

/**
 * This is the model class for table "af_addr_loan".
 *
 * @property string $id
 * @property string $request_id
 * @property integer $aid
 * @property string $user_id
 * @property integer $user_total
 * @property integer $loan_all
 * @property integer $overdue_norepay
 * @property integer $overdue_repay
 * @property integer $overdue7_norepay
 * @property integer $overdue7_repay
 * @property integer $loan_total
 * @property string $last_loan_day
 * @property integer $normal_repay
 * @property integer $realadl_tot_reject_num
 * @property integer $realadl_tot_freject_num
 * @property integer $realadl_tot_sreject_num
 * @property integer $realadl_tot_dlq14_num
 * @property integer $realadl_dlq14_ratio
 * @property integer $history_bad_status
 * @property string $create_time
 */
class AfAddrLoan extends BaseDBModel
{
    /**
     * @inheritdoc
     */
    public static function tableName()
    {
        return 'af_addr_loan';
    }

    /**
     * @return \yii\db\Connection the database connection used by this AR class.
     */
    public static function getDb()
    {
        return Yii::$app->get('db_antifraud');
    }

    /**
     * @inheritdoc
     */
    public function rules()
    {
        return [
            [['request_id', 'aid', 'user_id', 'user_total', 'loan_all', 'overdue_norepay', 'overdue_repay', 'overdue7_norepay', 'overdue7_repay', 'loan_total', 'last_loan_day', 'normal_repay', 'create_time'], 'required'],
            [['request_id', 'aid', 'user_id', 'user_total', 'loan_all', 'overdue_norepay', 'overdue_repay', 'overdue7_norepay', 'overdue7_repay', 'loan_total', 'normal_repay', 'realadl_tot_reject_num', 'realadl_tot_freject_num', 'realadl_tot_sreject_num', 'realadl_tot_dlq14_num', 'realadl_dlq14_ratio', 'history_bad_status'], 'integer'],
            [['last_loan_day', 'create_time'], 'safe']
        ];
    }

    /**
     * @inheritdoc
     */
    public function attributeLabels()
    {
        return [
            'id' => 'ID',
            'request_id' => '请求id',
            'aid' => '业务id',
            'user_id' => '用户ID',
            'user_total' => '通讯录是平台用户数量',
            'loan_all' => '通讯录与loan表总数(含申请)',
            'overdue_norepay' => '通讯录逾期未还款数量',
            'overdue_repay' => '通讯录逾期已还款数量',
            'overdue7_norepay' => '通讯录逾期7天未还款数量',
            'overdue7_repay' => '通讯录逾期7天已还款数量',
            'loan_total' => '通讯录通讯录有过放款数量',
            'last_loan_day' => '通讯录最近一次申请借款天数',
            'normal_repay' => '通讯录借款提前/正常还款',
            'realadl_tot_reject_num' => 'Realadl Tot Reject Num',
            'realadl_tot_freject_num' => 'Realadl Tot Freject Num',
            'realadl_tot_sreject_num' => 'Realadl Tot Sreject Num',
            'realadl_tot_dlq14_num' => 'Realadl Tot Dlq14 Num',
            'realadl_dlq14_ratio' => 'Realadl Dlq14 Ratio',
            'history_bad_status' => '通讯录中有过申请且历史最坏账单状态',
            'create_time' => '创建时间',
        ];
    }
}
